### PR TITLE
Display last event sessions on home screen

### PR DESCRIPTION
**Pull request by kleinMaggus**
I've added the last event sessions to the home screen for faster access.

Screenshot_Home_Last_Event

Please test it espacially with schedule and update every one minute stuff. And if you have a smart idea on how to populate the rows (HomeFragment:89) in an easier and extensible way you're welcome :D

Let me know what your thoughts are :)

### CODE REVIEW

**Comment by leonardoxh** [comment]
Hi, sorry for late reply, I am not sure about this feature itself, I think it is becoming a bit out of control on this part in terms of the simplicity I would like to keep.

**Comment by kleinMaggus** [comment]
Hi, np. I like the idea of having fast access directly from the home screen. But if you want to keep it simple, we can remove the PR. 
For sure I will create another one with the requestFocus for the teaser image because it does not have the focus when I start my app 😅

**Comment by nilsnilsnils** [comment]
I often watch the replay of the qualification or the race when I don't have time watching live. I would like to have fast access to current live session or quali and race replay of the last race. So 3 items would be enough for me. Its more important than old seasons. Therefore this pull request has my support.

I thank all of you for your great work :)

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/home/HomeActivity.kt [hunk 1]
```diff
@@ -2,13 +2,38 @@ package fr.groggy.racecontrol.tv.ui.home
...
 @AndroidEntryPoint
 class HomeActivity : FragmentActivity(R.layout.activity_home) {
     companion object {
+        private val TAG = HomeActivity::class.simpleName
+
         fun intent(context: Context) = Intent(context, HomeActivity::class.java)
     }
+
+    @Inject
+    lateinit var seasonService: SeasonService
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/home/HomeActivity.kt:26]: internal ?

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/home/HomeViewModel.kt [hunk 2]
```diff
@@ -4,13 +4,20 @@ import androidx.lifecycle.ViewModel
...
 import fr.groggy.racecontrol.tv.f1tv.Archive
+import fr.groggy.racecontrol.tv.ui.season.browse.Season
+import kotlinx.coroutines.flow.Flow
 import javax.inject.Inject
 
 @HiltViewModel
 class HomeViewModel @Inject constructor(
     private val seasonService: SeasonService
-): ViewModel() {
+) : ViewModel() {
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/home/HomeViewModel.kt:14]: I guess this small change can just be reverted

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/season/archive/SeasonArchiveFragment.kt [hunk 3]
```diff
@@ -9,13 +9,15 @@ import androidx.leanback.widget.*
...
+import fr.groggy.racecontrol.tv.ui.common.CustomListRowPresenter
 import fr.groggy.racecontrol.tv.ui.season.browse.SeasonBrowseActivity
 
 @Keep
 @AndroidEntryPoint
 class SeasonArchiveFragment : BrowseSupportFragment(), OnItemViewClickedListener {
 
-    private val archivesAdapter = ArrayObjectAdapter(ListRowPresenter())
+    private val listRowPresenter = CustomListRowPresenter()
+    private val archivesAdapter = ArrayObjectAdapter(listRowPresenter)
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/season/archive/SeasonArchiveFragment.kt:20]: Inline this as 

`
private val archivesAdapter = ArrayObjectAdapter(CustomListRowPresenter())
`

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/home/HomeFragment.kt [hunk 4]
```diff
@@ -70,8 +78,48 @@ class HomeFragment : RowsSupportFragment(), OnItemViewClickedListener {
...
+                .observe(viewLifecycleOwner, ::onUpdatedSeason)
+        }
+
+        archivesRow = getArchiveRow(viewModel)
+    }
+
+    private fun onUpdatedSeason(season: Season) {
+        val events = season.events.filter { it.sessions.isNotEmpty() }
+        if (events.isNotEmpty()) {
+            val event = events[0]
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/home/HomeFragment.kt:92]: I think there is a convenience method on arrays that is `events.first()`

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/home/HomeFragment.kt [hunk 5]
```diff
@@ -11,22 +11,29 @@ import androidx.core.view.updateLayoutParams
...
 @AndroidEntryPoint
 class HomeFragment : RowsSupportFragment(), OnItemViewClickedListener {
 
-    private val listRowPresenter = ListRowPresenter(FocusHighlight.ZOOM_FACTOR_NONE).apply {
-        shadowEnabled = false
-        selectEffectEnabled = false
-    }
-    private val archivesAdapter = ArrayObjectAdapter(listRowPresenter)
+    private val listRowPresenter = CustomListRowPresenter()
+    private val homeEntriesAdapter = ArrayObjectAdapter(listRowPresenter)
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/home/HomeFragment.kt:33]: Same (inline)

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/home/HomeFragment.kt [hunk 6]
```diff
@@ -11,22 +11,29 @@ import androidx.core.view.updateLayoutParams
...
+    private var imageView: ImageView? = null
+    private val currentYear = Year.now().value
+    private var archivesRow: ListRow? = null
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/home/HomeFragment.kt:36]: Is this variable used?

**Review by leonardoxh — CHANGES_REQUESTED** [review]
Ok, I have been thinking a lot and I am also in middle of job switching so pretty busy.

Hey I will test this PR this weekend and if all good let's merge it Monday (after the few changes requested)

**Pull request merged**